clients/fides-js/src/lib/preferences.ts

```diff
@@ -46,36 +45,26 @@ export const updateConsentPreferences = async ({
   debug?: boolean;
   servedNotices?: Array<LastServedNoticeSchema> | null;
   tcf?: TcfSavePreferences;
-  updateCookie?: (oldCookie: FidesCookie) => Promise<FidesCookie>;
+  updateCookie: (oldCookie: FidesCookie) => Promise<FidesCookie>;
 }) => {
-  // Derive the CookieKeyConsent object from privacy notices
-  const noticeMap = new Map<string, boolean>(
-    consentPreferencesToSave.map(({ notice, consentPreference }) => [
-      notice.notice_key,
-      transformUserPreferenceToBoolean(consentPreference),
-    ])
-  );
-  const consentCookieKey: CookieKeyConsent = Object.fromEntries(noticeMap);
```

Comment:
similar to https://github.com/ethyca/fides/pull/4131/files#r1333616066, this moves the cookie assignment logic out of the shared function, and into a notice-specific function. this is because otherwise, when we save TCF (which has no notices), we would overwrite `cookie.consent` to equal `{}` which isn't right!